rcl_yaml_param_parser/test/test_yaml_descriptor.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include "osrf_testing_tools_cpp/scope_exit.hpp"
+#include "../src/impl/yaml_descriptor.h"
+#include "rcutils/allocator.h"
+#include "rcutils/strdup.h"
+
+#define TEST_DESCRIPTOR_COPY(field, tmp_field) \
+  do { \
+    SCOPED_TRACE("TEST_DESCRIPTOR_COPY " #field); \
+    rcl_param_descriptor_t src_descriptor{}; \
+    rcl_param_descriptor_t dest_descriptor{}; \
+    rcutils_allocator_t allocator = rcutils_get_default_allocator(); \
+    src_descriptor.field = &tmp_field; \
```

Comment:
@bpwilcox nit: you may also do:
```suggestion
    auto tmp_field = field_value;
    src_descriptor.field = &tmp_field; \
```
to not have to add a temporary lvalue for each literal rvalue you want to use for testing.